### PR TITLE
feat(console): implement resource running state indicators

### LINKED ISSUE #6622: feat(console): show resource status
The intent of this change is to improve the Console diagnostics when a construct fails to initialize during inflight.

image

- The map view will show a small indicator for every construct that failed to initialize (e.g. red circle with exclamation mark), or that it's depending on a resource that failed (e.g. gray circle with a pause icon)
- The tree view will also show small indicators
- The inspector view will also show indicators
- The logs view will show the initialization errors

Also, the simulator should emit these events for every resource:

- starting, error, stopping, started, stopped

Proposal: #6593.

### PR DESCRIPTION
**Pull request by skyrpex**
Closes #6622.

### CODE REVIEW

File: libs/wingsdk/src/simulator/simulator.ts [hunk 1]
```diff
@@ -867,25 +919,36 @@ export class Simulator {
...
+    try {
+      const attrs = await resourceObject.init(context);
+
+      this.state[path].attrs = {
+        ...this.state[path].attrs,
+        ...attrs,
+      };
+    } catch (error: any) {
+      initError = error;
+    }
```
Comment (Chriscbr) [inline libs/wingsdk/src/simulator/simulator.ts:932]: I think there are other parts of the simulator expecting to catch and handle initialization errors - I wonder if they need to be updated somehow somehow to be compatible with this change

https://github.com/winglang/wing/blob/cdfab81a04acc494f84d01b74838e8cba72fb964/libs/wingsdk/src/simulator/simulator.ts#L279-L309

Comment (skyrpex) [reply libs/wingsdk/src/simulator/simulator.ts:932]: What I've seen is that constructs (e.g. resource.inflight.ts) swallowed their errors, emitted an error trace and returned an empty attrs object.

If we want to shut everything down if a construct fails to initialize, the UX may be impacted (e.g. no bucket can be inspected while any construct failed).

File: apps/wing-console/console/ui/src/features/map-view.tsx [hunk 2]
```diff
@@ -253,9 +274,15 @@ const ConstructNode: FunctionComponent<PropsWithChildren<ConstructNodeProps>> =
...
-                highlight && "outline-4 border-sky-400 dark:border-sky-500",
+                "outline outline-0",
+                !highlight &&
+                  !hasError &&
+                  "border-slate-200 dark:border-slate-800",
+                highlight && !hasError && "border-sky-400 dark:border-sky-500",
+                highlight && "outline-4",
+                !hasError && "outline-sky-200/50 dark:outline-sky-500/50",
+                hasError &&
+                  "border-red-500 dark:border-red-800 outline-red-200/50 dark:outline-red-500/50",
```
Comment (polamoros) [inline apps/wing-console/console/ui/src/features/map-view.tsx:285]: The contrast is not good for nodes with errors in dark mode.
image

Comment (skyrpex) [reply apps/wing-console/console/ui/src/features/map-view.tsx]: It's brighter in dark mode now.

**Comment by skyrpex** [comment]
I think I'm also going to bubble upwards loading statuses (e.g. a child is loading, so the parent will show loading as well).

**Comment by eladb** [comment]
image

In the side panel, instead of the red dot, can we add another "Label" called "Status" with a text status and a red dot?

**Comment by skyrpex** [comment]
@eladb something like this?

image

**Comment by eladb** [comment]
> @eladb something like this?
> 
> image

Yes! And it should be easy to click on this and see the errors (e.g. through the log filter when it will be implemented).

**Comment by eladb** [comment]
@skyrpex can you share a screen recording of how this looks with the "loading" animation? I don't see this in the preview environment.

**Comment by skyrpex** [comment]
> @skyrpex can you share a screen recording of how this looks with the "loading" animation? I don't see this in the preview environment.


https://github.com/winglang/wing/assets/1077520/fcce1c97-20b0-4597-859a-1c900ed22ebe

**Review by eladb — APPROVED** [review]

**Review by ainvoner — APPROVED** [review]

**Pull request merged**